### PR TITLE
WT-13173 Add a perf test to measure modify read performance

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Comment by quchenhao** [comment]
@Siddhartha8899 Can you look at this?

File: bench/wtperf/wtperf.c
Comment (quchenhao) [inline bench/wtperf/wtperf.c:174]: I feel we have a bug here. We need to write the nul terminator. @ddanderson

File: bench/wtperf/runners/modify-read-btree.wtperf
Comment (Siddhartha8899) [inline bench/wtperf/runners/modify-read-btree.wtperf:5]: is there a reason we need a range of sizes for keys and values?

**Review by Siddhartha8899 — APPROVED** [review]
LGTM, I will add this in charts once this PR is merged.

File: bench/wtperf/runners/modify-read-btree.wtperf
Comment (quchenhao) [reply bench/wtperf/runners/modify-read-btree.wtperf:5]: We want to append to the values to change the value size.

File: bench/wtperf/wtperf.c
Comment (ddanderson) [inline bench/wtperf/wtperf.c:485]: At the top of this function (`worker()`) there is a call to:
```
if ((ret = session->open_cursor(
               session, wtperf->uris[workload->table_index], NULL, NULL, &cursor)) != 0) {
```
that should not take place if `reopen_cursor` is true.

File: bench/wtperf/wtperf.c
Comment (ddanderson) [reply bench/wtperf/wtperf.c:174]: @quchenhao Nice find! That fix looks correct to me.

**Review by ddanderson — APPROVED** [review]
@quchenhao There's one item I flagged that needs to be fixed that would be a cursor leak. I'm marking this as approved as the remainder looks correct and I'm headed out to vacation, I don't want this to block you.

**Pull request merged**
